### PR TITLE
Add last_observed_time for raft messages

### PR DESCRIPTION
**Pull request by hhwyt**
For RaftMessage send & send wait duration metric, we need the `last_observed_time` field for the BatchRaftMessage. More details see: https://github.com/tikv/tikv/pull/17735.

### CODE REVIEW

File: proto/raft_serverpb.proto [hunk 1]
```diff
@@ -29,6 +29,8 @@ message RaftMessage {
     bytes extra_ctx = 11;
 
     disk_usage.DiskUsage disk_usage = 12;
+    // Used to measure the send wait duration.
+    uint64 last_observed_time = 13;
```
Comment (Connor1996) [inline proto/raft_serverpb.proto:33]: As `BatchRaftMessage` has one, why bother recording it here?

**Review by Connor1996 — APPROVED** [review]

**Review by v01dstar — APPROVED** [review]
lgtm

**Review by LykxSassinator — APPROVED** [review]
LGTM

**Comment by glorv** [comment]
@overvenus PTAL

**Review by glorv — APPROVED** [review]

File: proto/raft_serverpb.proto (hunk 1 above)
Comment (overvenus) [reply proto/raft_serverpb.proto:33]: Looks like we don't need to add a new field in protobuf, as step 1 occurs within the process with no inter-network communication.

**Review by overvenus — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Review by cfzjywxk — APPROVED** [review]
Better to update the latency break down whole picture
like https://docs.google.com/presentation/d/1Jsha70eEGUOWQOGR5VtDkplGMZYMBMmVh67At6o_4lk/edit#slide=id.geb29d9636c_2_0 does,
and update related information in https://docs.pingcap.com/tidb/stable/latency-breakdown

**Review by overvenus — APPROVED** [review]

**Comment by hhwyt** [comment]
@cfzjywxk PTAL again, the LGTM label was removed by a misoperation. Thanks~

**Review by v01dstar — APPROVED** [review]
lgtm

**Review by cfzjywxk — APPROVED** [review]

**Pull request merged**